src/points/activityCards/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import Touchable from 'src/components/Touchable'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { StyleSheet, Text, View } from 'react-native'
+import { Colors } from 'src/styles/colors'
+import CheckmarkWithCircleBorder from 'src/icons/CheckmarkWithCircleBorder'
+
+interface Props {
+  completed: boolean
+  icon: React.ReactNode
+  title: string
+  testID?: string
+  points?: number
+  onCardPress?: () => void
+}
+
+export default function ActivityCard({ completed, icon, title, onCardPress, testID }: Props) {
+  const containerStyle = {
+    ...styles.cardContainer,
+    ...(completed ? { opacity: 0.5 } : {}),
+  }
+
+  const content = (
+    <View style={containerStyle}>
+      {completed && (
+        <View style={styles.checkmarkIcon}>
+          <CheckmarkWithCircleBorder />
+        </View>
+      )}
+      {icon}
+      <Text style={styles.cardTitle}>{title}</Text>
+    </View>
+  )
+  if (onCardPress) {
+    return (
+      <Touchable testID={testID} style={styles.container} onPress={onCardPress}>
+        {content}
+      </Touchable>
+    )
+  } else {
+    return (
+      <View testID={testID} style={styles.container}>
```

Comment:
instead of a separate component per activity, could we have a config that contains all the customisable info and render everything together? i think this is in essence what you've done, but i'm surprised by how many layers and new files there are. naively i thought it'd be as straightforward as something like 
```
const pointsConfig = [{ // some props, like image, name }, ...]
...
render(
    <>
    ...some other stuff
    {pointsConfig.map(config => <ActivityCard {...config} />}
    </>
)
```

the activity card would then take the name and decide on the stuff that is customised in a predictable way (e.g. we could have something like the bottomSheetTitle is always ```t(`points.activityCards.${activityName}.bottomSheet.title`)```

Reply:
We should avoid prefer using static translation keys so we can eventually use static extraction.

https://valora-app.slack.com/archives/C025V1D6F3J/p1707472640807039

Reply:
@jeanregisser What are the requirements for this? Does this mean we can't store a translation key in a variable and then pass that to the translation function? I understand that we shouldn't dynamically generate the translation key _itself_ (i.e., they should be considered opaque) but can we safely do something like?

ts```
const translationKey = condition ? 'some.key' : 'some.other.key'
...
t(translationKey)
```

Reply:
yes almost, best is like this:

```ts
const translation = condition ? t('some.key') : t('some.other.key')
```